### PR TITLE
Add SQL server memory db

### PR DESCRIPTION
**Pull request by kbeaugrand**
## Motivation and Context (Why the change? What's the scenario?)
As discussed with @dluc I suggest my connector's code for SQL Server.

## High level description (Approach, Design)
Migrate code from https://github.com/kbeaugrand/SemanticKernel.Connectors.Memory.SqlServer to Kernel Memory

### CODE REVIEW

File: extensions/SQLServer/SQLServer/SqlServerMemory.cs [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+
+        if (!(await this.DoesIndexExistsAsync(index, cancellationToken).ConfigureAwait(false)))
+        {
+            // Index does not exist
+            yield break;
+        }
+
+        Embedding embedding = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
+
+        string queryColumns = "[id], [payload], [tags]";
```
Comment (dluc) [inline extensions/SQLServer/SQLServer/SqlServerMemory.cs:288]: looks like this var is not used in this method, can I remove it?

Comment (kbeaugrand) [reply extensions/SQLServer/SQLServer/SqlServerMemory.cs:288]: No, embedding is used as the vector search at line 353 on the same method.

Comment (dluc) [reply extensions/SQLServer/SQLServer/SqlServerMemory.cs:288]: sorry I mean the `queryColumns` var, it's not used, is it?

Comment (kbeaugrand) [reply extensions/SQLServer/SQLServer/SqlServerMemory.cs:288]: My bad, you're identifying a bug in fact. I should use this var but it's in fact hard coded and the end... please let me fix it.

**Review by dluc — APPROVED** [review]

**Pull request merged**